### DISCUSSION TITLE
How to convert string to Prisma.Decimal while storing it in my sql

### DISCUSSION THREAD

**Discussion by deepajais95** [prisma/prisma#9705] [Q&A] [answered]
eg: 
```prisma
model item{
  id          String  @db.Char(30)
  price     Decimal  @db.Decimal(10, 6)
  name    String   @db.VarChar(255)
  @@id([id])
}
```

```ts
interface  IItem{
id:string;
price: string;
name:string;
}
```

now if i want to store item in my sql
```ts
const item:IItem={
id:'1',
name:'Abc',
price:'12.23'
}
prisma.item.create({
 data:{
  id:item.id.
  name:item.name,
  price: ????????
 }
});
```
how to store price in sql

**Comment by ryands17** [comment, marked as answer]
@deepajais95 👋 

You can simply pass the string value and it will work. So `item.price`.

**Reply by deepajais95** [reply]
@ryands17 i have tried that only but its showing an error 
Type 'string' is not assignable to type 'Decimal'.
The expected type comes from property 'price' which is declared here on type 'item'
export type item={
id:string
name:string
price:Prisma.Decimal
}

**Reply by ryands17** [reply]
What version of `prisma` and `@prisma/client` are you on?

**Reply by deepajais95** [reply]
Both 3.1.1

**Reply by deepajais95** [reply]
updated to 3.2.1
and used this
const itemData: item={
  id:item.id.
  name:item.name,
  price: Decimal(item.price)
}
prisma.item.create({
 data:itemData
});
now its working

**Reply by ryands17** [reply]
Passing a string will also work in 3.2.1. The types specify either a string or Decimal that can be passed.

**Reply by deepajais95** [reply]
if i will use prisma type then it is not working eg: const itemData: item={price:'122'} because here item.price type is Decimal in schema.prisma

**Reply by ryands17** [reply]
Just checked with version `3.2.1`. This is the types that create accepts, so it's perfectly fine if you pass a string:

![image](https://user-images.githubusercontent.com/19697099/137733096-57013b0c-6794-49df-be73-cf1705fec6a6.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
